docs/content/latest/api/ysql/syntax_resources/ysql_grammar.ebnf

```diff
@@ -343,9 +343,17 @@ column_constraint ::= [ 'CONSTRAINT' constraint_name ]
 table_constraint ::= [ 'CONSTRAINT' constraint_name ]
                      ( 'CHECK' '(' expression ')'
                        | 'UNIQUE' '(' column_names ')' index_parameters
-                       | 'PRIMARY' 'KEY' '(' column_names ')'
+                       | 'PRIMARY' 'KEY' '(' key_columns ')'
                        | 'FOREIGN' 'KEY' '(' column_names ')' references_clause ) ;
 
+key_columns ::= hash_columns [ ',' range_columns ]
+              | range_columns ;
+
+hash_columns ::= column_name [ 'HASH' ]
```

Comment:
can we have `primary key (a HASH, b HASH, c HASH, d ASC ...)`? 
In that case maybe it should be:
```
hash_columns ::= ( column_name [ 'HASH' ] ) { ','  (column_name [ 'HASH' ] ) }
                 | '(' column_name { ',' column_name } ')' [ 'HASH' ] ;
```

Also I assume that something like `primary key ((a, b) HASH, c HASH)` or  `primary key (a HASH, (b,  c) HASH)` is not allowed.

Reply:
All of those are not allowed.
We only allow `(a HASH)` or `((a, b,...) HASH)`.